### ISSUE TITLE
How to print the gradient with respect to the output

### ISSUE THREAD

**Issue by xqustc** [Microsoft/CNTK#2577]
Hi, I write a loss function by self-define in python. How can I print the gradients from the user-defined loss function?

**Comment by cha-zhang** [comment]
Use cntk.ops.functions.Function.grad?

**Comment by xqustc** [comment]
Thanks @cha-zhang. I print the gradients using cntk.ops.functions.Function.grad. But the results differ from what I manually calculated. I do not know why.

**Comment by cha-zhang** [comment]
Could you provide a repro? We can take a look. Thanks!

**Comment by xqustc** [comment]
@cha-zhang Thanks for your reply. I want to know how CNTK computes the derivative of DNN output node given a user-defined loss function.  Here is a example. I modified the squared error loss function with weighting coefficients in different frequency bins. The following is my code:
z = fm(inputfea)
error = C.minus(label, z)
error_covar = C.element_times(error, factor)
se1 = C.element_times(error_covar, error)
se = C.reduce_sum(se1, -1)
train_loss = C.reduce_sum(se)
where z is the output of my network. Factor is weighting coefficients to be added in the train_loss function. In each training batch, factor is firstly calculated and then passed to the train_loss function to optimize network parameters. I use the following code to pass the factor,
batch_inputs2 = {inputfea:mb_data[inputfea].data, label:mb_data[label].data, factor:constant2}
In CNTK, we define a training loss, and the back-propagation procedure will be done. Now I want to know in my user-defined loss function, how does it compute the derivative of DNN output for back-propagation procedure.

**Comment by xqustc** [comment]
In squared_error loss function, the backward is calculated as: backward = 2 * np.subtract(o, t)
I want to know in my user-defined training loss, whether the backward is calculated as the following:
backward = 2 * np.subtract(o, t) * factor
Thanks!

**Comment by cha-zhang** [comment]
Gradient is computed with chain rule when you use a series of built-in ops to define your loss function. You shouldn't need to worry if the gradient is incorrect. You man feed some data to the loss and compute the gradient explicitly and verify, such as in this example: 
https://cntk.ai/pythondocs/cntk.ops.functions.html?highlight=grad#cntk.ops.functions.Function.grad
